daemon/endpoint.go

```diff
@@ -46,41 +46,42 @@ func (h *getEndpoint) Handle(params GetEndpointParams) middleware.Responder {
 	log.WithField(logfields.Params, logfields.Repr(params)).Debug("GET /endpoint request")
 
 	var wg sync.WaitGroup
+	var epListChan = make(chan int, 1)
 
 	if params.Labels == nil {
-		i := 0
 		endpointmanager.Mutex.RLock()
-		eps := make([]*models.Endpoint, len(endpointmanager.Endpoints))
+		eps := make([]*models.Endpoint, 0, len(endpointmanager.Endpoints))
 		wg.Add(len(endpointmanager.Endpoints))
-		for k := range endpointmanager.Endpoints {
-			go func(wg *sync.WaitGroup, i int, ep *endpoint.Endpoint) {
-				eps[i] = ep.GetModel()
+		for _, v := range endpointmanager.Endpoints {
+			go func(wg *sync.WaitGroup, ep *endpoint.Endpoint) {
+				<-epListChan
+				eps = append(eps, ep.GetModel())
```

Comment:
This is not working without a mutex.
I think the idea was to have a channel of models.
So do something like:
```
modelChan := make(chan *models.Endpoint, 1)
defer close(modelChan)
endpointmanager.Mutex.RLock()
numEndpoints := len(endpointmanager.Endpoints)
for _, v := range endpointmanager.Endpoints {
    go func(wg *sync.WaitGroup, ep *endpoint.Endpoint) {
        modelChan <- ep.GetModel()
    }(&wg, v)
}
endpointmanager.Mutex.RUnlock()
eps := make([]*models.Endpoint, 0, numEndpoints)
for i := 0; i < numEndpoints; i++ {
    eps = append(eps, <-modelChan)
}
return NewGetEndpointOK().WithPayload(eps)
```